AutoRest/Generators/CSharp/CSharp/TemplateModels/ModelTemplateModel.cs

```diff
@@ -201,39 +201,20 @@ private static string CreateSignature(IEnumerable<ConstructorParameterModel> par
             private static IEnumerable<string> CreateSignatureDocumentation(IEnumerable<ConstructorParameterModel> parameters)
             {
                 var declarations = new List<string>();
-                IEnumerable<ConstructorParameterModel> parametersWithDocumentation =
-                    parameters.Where(p => !(string.IsNullOrEmpty(p.UnderlyingProperty.Summary) && string.IsNullOrEmpty(p.UnderlyingProperty.Documentation)));
 
-                foreach (var property in parametersWithDocumentation.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
-                {
-                    string documentation = string.Empty;
-                    string formatString = "<param name=\"{0}\">{1}</param>";
+                IEnumerable<Property> parametersWithDocumentation =
+                   parameters.Where(p => !(string.IsNullOrEmpty(p.UnderlyingProperty.Summary) &&
+                   string.IsNullOrEmpty(p.UnderlyingProperty.Documentation)) &&
+                   !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty);
 
-                    if (string.IsNullOrEmpty(property.Documentation))
-                    {
-                        documentation = string.Format(
-                            CultureInfo.InvariantCulture,
-                            formatString,
-                            char.ToLower(property.Name[0], CultureInfo.InvariantCulture) + property.Name.Substring(1),
-                            string.Empty);
-                    }
-                    else
-                    {
-                        documentation = string.Format(
-                            CultureInfo.InvariantCulture,
-                            formatString,
-                            char.ToLower(property.Name[0], CultureInfo.InvariantCulture) + property.Name.Substring(1),
-                            property.Documentation);
-                    }
-
-                    if (!string.IsNullOrEmpty(property.Summary))
-                    {
-                        documentation = string.Format(
-                            CultureInfo.InvariantCulture,
-                            formatString,
-                            char.ToLower(property.Name[0], CultureInfo.InvariantCulture) + property.Name.Substring(1),
-                            property.Summary);
-                    }
+                foreach (var property in parametersWithDocumentation)
+                {
+                    var documentationInnerText = property.Summary ?? property.Documentation;
+                    var documentation = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "<param name=\"{0}\">{1}</param>",
+                        char.ToLower(property.Name[0], CultureInfo.InvariantCulture) + property.Name.Substring(1),
+                        documentationInnerText);
```

Comment:
I think it's fine because in my linq query, I check for NullOrEmpty for the summary/description. There has to be some documentation. Otherwise, it will just omit it.